frontend/public/components/utils/error-boundary.tsx

```diff
@@ -30,12 +46,14 @@ export const withFallback: WithFallback = (Component, FallbackComponent) => (pro
   <Component {...props} />
 </ErrorBoundary>;
 
-export type WithFallback = <P = {}>(Component: React.ComponentType<P>, FallbackComponent?: React.ComponentType) => React.ComponentType<P>;
+export type WithFallback = <P = {}>(Component: React.ComponentType<P>, FallbackComponent?: React.ComponentType<ErrorBoundaryFallbackComponentProps>) => React.ComponentType<P>;
 
 export type ErrorBoundaryProps = {
-  FallbackComponent?: React.ComponentType;
+  FallbackComponent?: React.ComponentType<ErrorBoundaryFallbackComponentProps>;
```

Comment:
We are going to require that *every* fallback component uses these props?

Reply:
@alecmerdler probably not a good idea, so React.ComponentType<any> to broaden that restriction, right?

Reply:
@alecmerdler oh yes. This is all the possible data that a Component could use to react to the Error Boundary.